frontend/src/store/objectStore.ts

```diff
@@ -103,15 +111,104 @@ export const useObjectStore = defineStore('objectStore', () => {
   //   selectedObject.value = response.data;
   // }
 
+  async function getObjectPermissions(objectId: string) {
+    try {
+      loading.value = true;
+
+      const objPerms = (
+        await permissionService.objectGetPermissions(objectId)
+      ).data;
+
+      if (objPerms.length) {
+        // Get the user records for the unique user IDs in the perms
+        const uniqueIds = [...new Set(objPerms.map((x: any) => x.userId))].join(',');
+        const uniqueUsers = (await userService.searchForUsers({ userId: uniqueIds })).data;
+
+        const hasPermission = (userId: string, permission: string) => {
+          return objPerms.some((perm: any) => perm.userId === userId && perm.permCode === permission);
+        };
+
+        const userPermissions: UserPermissions[] = [];
+        uniqueUsers.forEach((user: User) => {
+          userPermissions.push({
+            userId: user.userId,
+            fullName: user.fullName,
+            read: hasPermission(user.userId, Permissions.READ),
+            update: hasPermission(user.userId, Permissions.UPDATE),
+            delete: hasPermission(user.userId, Permissions.DELETE),
+            manage: hasPermission(user.userId, Permissions.MANAGE),
+          });
+        });
+
+        selectedObjectPermissions.value = userPermissions;
+      } else {
+        selectedObjectPermissions.value = [];
+      }
+    }
+    finally {
+      loading.value = false;
+    }
+  }
+
+  async function addObjectPermission(
+    bucketId: string,
+    userId: string,
+    permCode: string
+  ) {
+    try {
+      loading.value = true;
+
+      await objectService.addPermissions(bucketId, [{ userId, permCode }]);
+    } finally {
+      loading.value = false;
+    }
+  }
+
+  async function deleteObjectPermission(
+    bucketId: string,
+    userId: string,
+    permCode: string
+  ) {
+    try {
+      loading.value = true;
+
+      await objectService.deletePermission(bucketId, { userId, permCode });
+    } finally {
+      loading.value = false;
+    }
+  }
+
+  async function removeObjectUser(bucketId: string, userId: string) {
+    try {
+      loading.value = true;
+
+      for (const [, value] of Object.entries(Permissions)) {
+        await objectService.deletePermission(bucketId, {
+          userId,
+          permCode: value,
+        });
+      }
```

Comment:
Two thoughts:
1) If we're only using the values and don't need the key/value pairing, we should probably be using `Object.values()` instead.
```ts
for (const value of Object.values(Permissions)) {
```
2) Since this looks like a general async dispatcher pattern, it may be better to wrap this into a `Promise.all()` for synchronization purposes:
```ts
await Promise.all(
  Object.values(Permissions)
    .map(perm => objectService.deletePermission(bucketId, {
      userId,
      permCode: value,
    }));
);
```

Reply:
Used the same logic as in existing bucketStore removeBucketUser.
Think there's possibly a way to take in the service method and have this permission logic extracted to a common util function (for buckets and objects). So someone could take a look at that at some point and implement your logic above for both (ideally in one palce)